utils/benchmarking/eventing/tscli/main.go

```diff
@@ -0,0 +1,131 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	ctrdlog "github.com/containerd/containerd/log"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"eventing/proto"
+)
+
+func usage() {
+	logrus.Errorf(
+		"Usage: %[1]s <address> start <definition.json>\n"+
+			"Usage: %[1]s <address> end   <results.json>\n",
+		os.Args[0],
+	)
+}
+
+func main() {
+	logrus.SetFormatter(&logrus.TextFormatter{
+		TimestampFormat: ctrdlog.RFC3339NanoFixed,
+		FullTimestamp:   true,
+	})
+
+	if len(os.Args) != 4 {
+		usage()
+		os.Exit(2)
+	}
+
+	address := os.Args[1]
+
+	// Assuming that they won't use IPv6...
+	if !strings.ContainsRune(address, ':') {
+		logrus.Fatalln("missing the port number in parameter <address>")
+	}
+
+	conn, err := grpc.Dial(address, grpc.WithInsecure(), grpc.WithBlock())
+	if err != nil {
+		logrus.Fatalln("failed to dial", err)
+	}
+	defer conn.Close()
+	timeseriesClient := proto.NewTimeseriesClient(conn)
+
+	switch os.Args[2] {
+	case "start":
+		start(timeseriesClient)
+	case "end":
+		end(timeseriesClient)
+	default:
+		usage()
+		os.Exit(2)
+	}
+}
+
+func start(client proto.TimeseriesClient) {
+	data, err := ioutil.ReadFile(os.Args[3])
+	if err != nil {
+		logrus.Fatalln("failed to readfile", err)
+	}
+	var experimentDefinition proto.ExperimentDefinition
+	if err := json.Unmarshal(data, &experimentDefinition); err != nil {
+		logrus.Fatalln("failed to unmarshal", err)
+	}
+	if _, err := client.StartExperiment(context.Background(), &experimentDefinition); err != nil {
+		logrus.Fatalln("failed to start experiment", err)
+	}
+	logrus.Infoln("started experiment")
+}
+
+func end(client proto.TimeseriesClient) {
+	res, err := client.EndExperiment(context.Background(), &empty.Empty{})
+	if err != nil {
+		logrus.Fatalln("failed to end experiment", err)
+	}
+
+	content, err := json.MarshalIndent(res, "", "\t")
+	if err != nil {
+		logrus.Fatalln("failed to marshal", err)
+	}
+	if err := ioutil.WriteFile(os.Args[3], content, 0666); err != nil {
+		logrus.Fatalln("failed to writefile", err)
+	}
+
+	for _, invocation := range res.Invocations {
+		logrus.Infof(
+			`INVOCATION {%s}
+Id       : %s
+InvokedOn: %s
+Duration : %dμs
+Status   : %s
+`,
```

Comment:
how about using the CSV format? it's much more friendly to inspect in Excel/Sheet or even eyeball